pkg/config/jobs_test.go

```diff
@@ -0,0 +1,118 @@
+package config
+
+import (
+	"k8s.io/apimachinery/pkg/util/diff"
+	"reflect"
+	"testing"
+
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+func TestPresubmitsAddAll(t *testing.T) {
+	testCases := []struct {
+		description string
+		source      Presubmits
+		destination Presubmits
+		expected    Presubmits
+	}{{
+		description: "merge empty structure into empty structure",
+	}, {
+		description: "merge empty structure into non-empty structure",
+		source:      Presubmits{},
+		destination: Presubmits{"org/repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "destination-job-for-org-repo"}},
+		}},
+		expected: Presubmits{"org/repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "destination-job-for-org-repo"}},
+		}},
+	}, {
+		description: "merge non-empty structure into empty structure",
+		source: Presubmits{"org/repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "source-job-for-org-repo"}},
+		}},
+		destination: Presubmits{},
+		expected: Presubmits{"org/repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "source-job-for-org-repo"}},
+		}},
+	}, {
+		description: "merge different jobs for a single repo, result should have both",
+		source: Presubmits{"org/repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "source-job-for-org-repo"}},
+		}},
+		destination: Presubmits{"org/repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "destination-job-for-org-repo"}}}},
+		expected: Presubmits{"org/repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "destination-job-for-org-repo"}},
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "source-job-for-org-repo"}},
+		}},
+	}, {
+		description: "merge jobs for different repos, result should have both",
+		source: Presubmits{"org/source-repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "source-job-for-org-source-repo"}},
+		}},
+		destination: Presubmits{"org/destination-repo": {
+			prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "destination-job-for-org-destination-repo"}}}},
+		expected: Presubmits{
+			"org/source-repo":      {prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "source-job-for-org-source-repo"}}},
+			"org/destination-repo": {prowconfig.Presubmit{JobBase: prowconfig.JobBase{Name: "destination-job-for-org-destination-repo"}}},
+		},
+	}, {
+		description: "merge jobs with same name for a single repo, result has the one originally in destination",
```

Comment:
Since we need unique names to work this might be better as an error, but meh

Reply:
It cannot be an error. In rehearse, when both ci-operator config and a presubmit that uses it change, we'll want to rehearse the same job for two reasons, so it ends up being returned by two different "what jobs do we need to rehearse" methods, and those results need to be merged.

Reply:
Ah